File: clojure/bob/src/bob.clj
(ns bob
  (:require [clojure.string :as str]))

(defn containsLetters [talk]
  (re-find (re-matcher #"[a-zA-Z]+" talk))
)

(defn lettersAllUppercase [talk]
  (= (str/upper-case talk) talk)
)

(defn shouting [talk]
  (and (containsLetters talk) (lettersAllUppercase talk))
)

(defn question [talk]
  (str/ends-with? talk "?")
)

(defn silence [talk]
  (str/blank? talk)
)

(defn response-for [talk]
  (if (silence talk)
    "Fine. Be that way!"
    (if (shouting talk)
      "Whoa, chill out!"
      (if (question talk)
        "Sure."
        "Whatever."
      )
    )
  )
)

File: javascript/bob/bob.js
const silence = talk => talk.trim().length === 0
const yelling = talk => talk.toUpperCase() === talk && talk.toLowerCase() !== talk
const question = talk => talk.endsWith('?')

module.exports = function () {
  this.hey = talk => {
    let response = 'Whatever.'
    if (silence(talk)) response = 'Fine. Be that way!'
    else if (yelling(talk)) response = 'Whoa, chill out!'
    else if (question(talk)) response = 'Sure.'
    return response
  }
}
